### PR TITLE
set initial zoom level on cache map to include its waypoints (fix #7949)

### LINKED ISSUE #7949: Initial zoom on cache map not covering its waypoints
**Describe the bug:**
If a user opens the map for a cache, he expects to see all relevant information initially, but the opened map uses the last zoom level or some other preference which might not show all waypoints in the viewport.

**To Reproduce:**
1. Open a saved cache with additional waypoints (maybe spread over a wider area)
2. Click on the map icon to open the map for this cache

**Actual behavior/state after performing these steps:**
The map is opened centered on the listing coordinates. Additional waypoints of this cache might or might not be within the shown viewport (not sure where it depends on).

**Expected behavior/state after performing these steps:**
The viewport and zoomlevel of the cache map should be chosen in a way to show all cache waypoints.

**Version of c:geo used:**
2019.01.05

**Is the problem reproducible:**
Yes, all map types.

### PR DESCRIPTION
**Pull request by moving-bits**
If a map gets opened from a single cache: Set initial zoom level on map to include all of cache's waypoints (fixes #7949)

### CODE REVIEW

File: main/src/cgeo/geocaching/maps/CGeoMap.java [hunk 1]
```diff
@@ -456,7 +453,8 @@ protected void initializeMap(final ArrayList<Location> trailHistory) {
             followMyLocation = false;   // do not center on GPS position, even if in LIVE mode
         }
 
-        overlayPositionAndScale = mapView.createAddPositionAndScaleOverlay(getIntentCoords(), mapOptions.geocode);
+        Viewport viewport = getIntentViewport();
```
Comment (Lineflyer) [inline main/src/cgeo/geocaching/maps/CGeoMap.java]: Codacy Issue found: [Variable 'viewport' should be declared final.](https://app.codacy.com/manual/cgeo/cgeo/pullRequest?prid=4742102)

**Comment by Lineflyer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- main/src/cgeo/geocaching/location/Viewport.java  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/cgeo/cgeo/pullRequest?prid=4742102&bid=15825883)

File: main/src/cgeo/geocaching/maps/mapsforge/MapsforgeMapView.java [hunk 2]
```diff
@@ -183,6 +183,12 @@ public int getMapZoomLevel() {
         return getMapPosition().getZoomLevel() + 1;
     }
 
+    @Override
+    public void zoomToBounds(final Viewport bounds, GeoPointImpl center) {
```
Comment (Lineflyer) [inline main/src/cgeo/geocaching/maps/mapsforge/MapsforgeMapView.java]: Codacy Issue found: [Parameter center should be final.](https://app.codacy.com/manual/cgeo/cgeo/pullRequest?prid=4742102)

Comment (moving-bits) [reply main/src/cgeo/geocaching/maps/mapsforge/MapsforgeMapView.java]: Hmph, I thought I had called CheckStyle for all changed files... Should be somehow integrated in the local build process automatically. Anyway: Thanks for catching.

**Comment by Lineflyer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- main/src/cgeo/geocaching/location/Viewport.java  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/cgeo/cgeo/pullRequest?prid=4742102&bid=15825883)

**Comment by Lineflyer** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- main/src/cgeo/geocaching/location/Viewport.java  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/cgeo/cgeo/pullRequest?prid=4742102&bid=15825883)

**Pull request merged**